hs_file_types/models/netcdf.py

```diff
@@ -580,6 +580,12 @@ def set_file_type(cls, resource, file_id, user):
                         if k == 'subject':
                             logical_file.metadata.keywords = v
                             logical_file.metadata.save()
+                            # update resource level keywords
+                            resource_keywords = [subject.value.lower() for subject in
+                                                 resource.metadata.subjects.all()]
+                            for kw in logical_file.metadata.keywords:
+                                if kw.lower() not in resource_keywords:
+                                    resource.metadata.create_element('subject', value=kw)
```

Comment:
@pkdash Has this case (i.e., make sure no keyword duplication on resource level keywords) been captured in tests? I cannot tell from the code changes for this PR.